astropy/modeling/core.py

```diff
@@ -3089,6 +3089,14 @@ def inputs_map(self):
                         inputs_map[inp] = r_inputs_map[self.right.inputs[i - len(self.left.inputs)]]
                     else:
                         inputs_map[inp] = self.right, self.right.inputs[i - len(self.left.inputs)]
+        elif self.op == 'fix_inputs':
+            fixed_ind = list(self.right.keys())
+            ind = [list(self.left.inputs).index(i) if isinstance(i, str) else i for i in fixed_ind]
+            inp_ind = list(range(self.left.n_inputs))
+            for i in ind:
```

Comment:
Don't these indices need to be removed in reverse order?

Reply:
I'll do it to be on the safe side. But doesn't python now have some magic to deal with this issue?